src/step-function.ts

```diff
@@ -915,17 +939,28 @@ export type SyncExecutionResult<T> =
   | SyncExecutionFailedResult
   | SyncExecutionSuccessResult<T>;
 
-export interface ExpressStepFunction<F extends AnyFunction> {
-  (...args: Parameters<F>): SyncExecutionResult<ReturnType<F>>;
+type ConditionalMachine<P, O> = P extends undefined
```

Comment:
How about just replace this with the entire `AWS.StepFunctions.StartSyncExecution` type, except substitute `Input` with the user's type?

Reply:
Nevermind, I think we need to handle all of the paths...

Creating the variable signature in typescript is messy and I can't get it to work... also leaves the string parameters ambiguous.

Reply:
Going to start without support for the object input form, that way we can build time check the keys, can add the object form later.